scene/gui/color_picker.cpp

```diff
@@ -857,10 +960,14 @@ void ColorPicker::add_preset(const Color &p_color) {
 		_add_preset_button(_get_preset_size(), p_color);
 	}
 
+	palette_name->set_text(vformat("%s*", palette_name->get_text().replace("*", "")));
+	palette_name->set_tooltip_text(ETR("The changes to this palette have not been saved to a file."));
+
 #ifdef TOOLS_ENABLED
 	if (editor_settings) {
 		PackedColorArray arr_to_save = get_presets();
 		editor_settings->call(SNAME("set_project_metadata"), "color_picker", "presets", arr_to_save);
```

Comment:
You could do something like `const StringName set_project_metadata = SNAME("set_project_metadata")`, to avoid making 2 SNAMES (each one separately allocates memory).